packages/ra-core/src/sideEffect/debounce.js

```diff
@@ -0,0 +1,58 @@
+import { fork, put, takeEvery, take } from 'redux-saga/effects';
+
+const tasks = {};
+
+// use the whole stringified payload to ensure actions such as getMatching
+// are not mistaken for another getMatching targeting another resource or
+// the same resource but with different filters, etc.
+const getKey = payload => JSON.stringify(payload);
+
+export const handleFinalize = key => {
+    delete tasks[key];
+};
+
+/**
+ * For example, will match `GET_MATCHING_SUCCESS` or `GET_MATCHING_FAILURE`
+ * if the specified type is `GET_MATCHING` and the action's requestPayload
+ * matches the key after being stringified
+ */
+const defaultIsRelatedAction = (type, key) => action =>
+    [`${type}_SUCCESS`, `${type}_FAILURE`].includes(action.type) &&
+    key === getKey(action.requestPayload);
+
+/**
+ * Dispatch the first action and ensure we don't prevent future actions of the same
+ * type with the same parameter to execute again by listening to the action
+ * `SUCCESS` or `FAILURE` counterparts and removing the action from our debounced tasks
+ * list
+ */
+export function* handleDebouncedAction(
+    key,
+    debouncedAction,
+    isRelatedAction = defaultIsRelatedAction(debouncedAction.type, key)
+) {
+    yield put(debouncedAction);
+    yield take(isRelatedAction, handleFinalize, key);
```

Comment:
I don't agree with the strategy. The debounce should be action agnostic, i.e. be usable even with non-`fetch` actions, and debounce not based on the response, but based on a delay.

Reply:
So you mean we should pass the key in the action itself (meta or payload) ?